### PR TITLE
Add Token Signal Box (#176)

### LINKED ISSUE #176: Add a signal box version of the token signals
### Is there an existing feature request for this?

- [X] I have searched the existing issues

### Describe what you would like added or changed

There is currently no signal box counterpart to the token signal. A signal box version could be used at exits and the regular token signal at entrances so an intersection can always show the correct signal to trains.

### PR DESCRIPTION
**Pull request by robotman2412**
**The Issue**
There is currently a signal box that acts like the block signal; the block signal relay box. There is not such a thing for token signals (#176).
 
**The Proposal**
This adds the Token Signal Box, which implements the functionality of the token signal in an ActiveSignalBox.
 
**Possible Side Effects**
There is now a second use of the Radio Chip item.
 
**Alternatives**
There were no alternative ways to add this signal box.

### CODE REVIEW

File: src/main/java/mods/railcraft/world/level/block/RailcraftBlocks.java [hunk 1]
```diff
@@ -33,20 +33,7 @@
...
-import mods.railcraft.world.level.block.signal.DualTokenSignalBlock;
-import mods.railcraft.world.level.block.signal.SignalBlockRelayBoxBlock;
-import mods.railcraft.world.level.block.signal.SignalBoxBlock;
-import mods.railcraft.world.level.block.signal.SignalCapacitorBoxBlock;
-import mods.railcraft.world.level.block.signal.SignalControllerBoxBlock;
-import mods.railcraft.world.level.block.signal.SignalInterlockBoxBlock;
-import mods.railcraft.world.level.block.signal.SignalReceiverBoxBlock;
-import mods.railcraft.world.level.block.signal.SignalSequencerBoxBlock;
-import mods.railcraft.world.level.block.signal.TokenSignalBlock;
+import mods.railcraft.world.level.block.signal.*;
```
Comment (Edivad99) [inline src/main/java/mods/railcraft/world/level/block/RailcraftBlocks.java]: Do not group imports

File: src/main/java/mods/railcraft/world/level/block/entity/RailcraftBlockEntityTypes.java [hunk 2]
```diff
@@ -25,19 +25,7 @@
...
-import mods.railcraft.world.level.block.entity.signal.DualBlockSignalBlockEntity;
-import mods.railcraft.world.level.block.entity.signal.DualDistantSignalBlockEntity;
-import mods.railcraft.world.level.block.entity.signal.DualTokenSignalBlockEntity;
-import mods.railcraft.world.level.block.entity.signal.SignalCapacitorBoxBlockEntity;
-import mods.railcraft.world.level.block.entity.signal.SignalControllerBoxBlockEntity;
-import mods.railcraft.world.level.block.entity.signal.SignalInterlockBoxBlockEntity;
-import mods.railcraft.world.level.block.entity.signal.SignalReceiverBoxBlockEntity;
-import mods.railcraft.world.level.block.entity.signal.SignalSequencerBoxBlockEntity;
-import mods.railcraft.world.level.block.entity.signal.TokenSignalBlockEntity;
+import mods.railcraft.world.level.block.entity.signal.*;
```
Comment (Edivad99) [inline src/main/java/mods/railcraft/world/level/block/entity/RailcraftBlockEntityTypes.java]: Do not group imports

File: src/main/java/mods/railcraft/world/level/block/entity/RailcraftBlockEntityTypes.java [hunk 3]
```diff
@@ -293,12 +281,18 @@ public static void register(IEventBus modEventBus) {
           () -> BlockEntityType.Builder
               .of(SignalReceiverBoxBlockEntity::new, RailcraftBlocks.SIGNAL_RECEIVER_BOX.get())
               .build(null));
-
+
```
Comment (Edivad99) [inline src/main/java/mods/railcraft/world/level/block/entity/RailcraftBlockEntityTypes.java]: Run `./gradlew spotlessApply`

File: src/main/resources/assets/railcraft/textures/entity/signal_box/token_signal_box.xcf
Comment (Edivad99) [inline src/main/resources/assets/railcraft/textures/entity/signal_box/token_signal_box.xcf:1]: Remove this file

File: src/main/java/mods/railcraft/world/level/block/RailcraftBlocks.java (hunk 1 above)
Comment (robotman2412) [reply src/main/java/mods/railcraft/world/level/block/RailcraftBlocks.java]: Split into class imports

File: src/main/java/mods/railcraft/world/level/block/entity/RailcraftBlockEntityTypes.java (hunk 2 above)
Comment (robotman2412) [reply src/main/java/mods/railcraft/world/level/block/entity/RailcraftBlockEntityTypes.java]: Split into class imports

File: src/main/resources/assets/railcraft/textures/entity/signal_box/token_signal_box.xcf
Comment (robotman2412) [reply src/main/resources/assets/railcraft/textures/entity/signal_box/token_signal_box.xcf:1]: Removed

File: src/main/java/mods/railcraft/world/level/block/entity/RailcraftBlockEntityTypes.java (hunk 3 above)
Comment (robotman2412) [reply src/main/java/mods/railcraft/world/level/block/entity/RailcraftBlockEntityTypes.java]: Done

**Comment by Edivad99** [comment]
I'll take a look tomorrow

**Pull request merged**